tiledb/sm/query/query_condition.h

```diff
@@ -319,29 +327,27 @@ class QueryCondition {
   /* ********************************* */
   /*         PRIVATE ATTRIBUTES        */
   /* ********************************* */
+  /** AST Tree structure representing the condition. **/
+  tdb_unique_ptr<tiledb::sm::ASTNode> tree_{};
 
-  /**
-   * All clauses in this condition. Clauses `clauses_[i]` and
-   * `clauses_[i + 1]` are combined by the `combination_ops_[i]`
-   * operator.
-   */
-  std::vector<Clause> clauses_;
+  /** Caches all clauses in `tree_`. */
+  mutable std::vector<Clause> clauses_;
 
-  /** Caches all field names in `clauses_`.  */
-  mutable std::unordered_set<std::string> field_names_;
+  /** Caches the logical operators to combine `clauses_`. */
+  mutable std::vector<QueryConditionCombinationOp> combination_ops_;
 
-  /** Logical operators to combine clauses stored in `clauses_`. */
-  std::vector<QueryConditionCombinationOp> combination_ops_;
+  /** Caches all field names in the value nodes of the AST.  */
+  mutable std::unordered_set<std::string> field_names_;
```

Comment:
Are these caches useful once we have an AST?

Reply:
They are useful for serialization ATM.